wps/core/models.py

```diff
@@ -16,20 +16,92 @@
 """Defines dataclasses for business-logic data as well as request/reply models for use
 in the API."""
 
-from pydantic import BaseModel
+from dataclasses import dataclass
+from enum import Enum
 
+from ghga_service_commons.utils.utc_dates import DateTimeUTC
+from pydantic import BaseModel, EmailStr, Field
 
-class WorkPackageData(BaseModel):
+__all__ = [
+    "WorkType",
+    "WorkOrderToken",
+    "WorkPackageCreationData",
+    "WorkPackageData",
+    "WorkPackage",
+]
+
+
+class BaseDto(BaseModel):
+    """Base model preconfigured for use as Dto."""
+
+    class Config:  # pylint: disable=missing-class-docstring
+        extra = "forbid"
+        frozen = True
+
+
+class WorkType(str, Enum):
+    """The type of work that a work package describes."""
+
+    DOWNLOAD = "download"
+    UPLOAD = "upload"
+
+
+@dataclass(frozen=True)
```

Comment:
Why is that not a pydantic model?

Reply:
It's only used internally as a way to bundle the data. Would you suggest to use Pydantic everywhere?

Reply:
I think for consistency it would be fine to use pydantic everywhere. There are also some nice synergies. A problem is, that is not entirely clear from the beginning on which models will be at the boundaries and which not. I think the performance overhead is neglectable. It is also not a major problem. If you like to discuss that in further depth, we can add it to the refactoring list.

Reply:
Agree. I think I will change that, but as part of the next PR.